### DISCUSSION TITLE
Letsencrypt staging

### DISCUSSION THREAD

**Discussion by snarum** [requarks/wiki#4653] [Help / Questions]
Hi

Does anyone know if there is a way to use letsencrypt staging environment in wiki.js? I'm having some issues with renewal of the letsencrypt certificate and now I have exhausted my 5 certificates a week limit...

**Comment by ieaster1** [comment]
As far as I can tell, you can accomplish this in dev mode:

https://github.com/Requarks/wiki/blob/c6933a2d20d30780d66729c3e26d7f03f8509adc/server/core/letsencrypt.js#L13

I would review https://docs.requarks.io/en/dev and see if it works for you.  I have not tested it, however.

**Reply by snarum** [reply]
Thanks. The problem I was experiencing was because I tried to renew the certificate from wiki.js admin gui too many times. Now I realize that wiki.js won't renew the certificate before it is 5 days left. I think til is too late, as letsencrypt.org recommmend renewing when its 1/3 of the period left. I'll make a suggestion to change the script.
